### PR TITLE
Cosmetics

### PR DESCRIPTION
**Pull request by davepwsmith**
Change 'page' to 'p' which is more usual in english.
This is, really, just my personal preference for formatting pandoc citations, but I think it is neater without the 'p.' or 'page' (probably because I use MLA style!)

### CODE REVIEW

File: chrome/content/zotero-better-bibtex/cayw.coffee [hunk 1]
```diff
@@ -102,7 +102,7 @@ class Zotero.BetterBibTeX.CAYW.CitationEditInterface
     # "label":"line","locator":"xx"
     citations = []
     for citation in @citation.citationItems
-      citation.label = 'page' if !citation.label && citation.locator
+      citation.label = 'p.' if !citation.label && citation.locator
```
Comment (retorquere) [inline chrome/content/zotero-better-bibtex/cayw.coffee]: This breaks the "scannable cite" translator (the label is used as a lookup)

Comment (davepwsmith) [inline chrome/content/zotero-better-bibtex/cayw.coffee]: Ah OK, no worries then! 

On 3 September 2015 14:10:54 BST, retorquere <notifications@github.com> wrote:
>> @@ -102,7 +102,7 @@ class
>Zotero.BetterBibTeX.CAYW.CitationEditInterface
>>      # "label":"line","locator":"xx"
>>      citations = []
>>      for citation in @citation.citationItems
>> -      citation.label = 'page' if !citation.label && citation.locator
>> +      citation.label = 'p.' if !citation.label && citation.locator
>
>This breaks the "scannable cite" translator (the label is used as a
>lookup)
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/ZotPlus/zotero-better-bibtex/pull/339/files#r38643160

--
https://keybase.io/davidsmith
--
Sent from my phone. Please excuse my brevity.

File: chrome/content/zotero-better-bibtex/cayw.coffee [hunk 2]
```diff
@@ -186,7 +186,7 @@ Zotero.BetterBibTeX.CAYW.Formatter = {
       cite += "#{citation.prefix} " if citation.prefix
       cite += '-' if citation['suppress-author']
       cite += "@#{citation.citekey}"
-      cite += ", #{citation.label} #{citation.locator}" if citation.locator
+      cite += " #{citation.locator}" if citation.locator
```
Comment (retorquere) [inline chrome/content/zotero-better-bibtex/cayw.coffee]: http://pandoc.org/demo/example19/Extension-citations.html (see "Citations go inside square brackets and are separated by semicolons.") includes the comma, are you sure about this?

**Comment by retorquere** [comment]
Huh, doesn't pandoc filter out the locator when the style requires it not to be there? That seems like a flaw in pandoc. But in any case, perhaps it could just translate "p." to "" *within* the pandoc formatter?

File: chrome/content/zotero-better-bibtex/cayw.coffee (hunk 1 above)
Comment (retorquere) [inline chrome/content/zotero-better-bibtex/cayw.coffee]: Doesn't mean we couldn't also (re)use a lookup inside the pandoc formatter. But this "if" just mends the Zotero picker, which is a bit whacky; it returns a locator label (kind, really) for every kind of locator you choose, except for "page". What that should display is a later case, it's more an identifier than a label at that point in the code.

**Comment by davepwsmith** [comment]
Hm, looks like I've been using pandoc citations differently to everyone else all this time. OK, so whether or not the word `page` or the comma or whatever is there, pandoc will filter it out to the correct output for any of the following cases:
`[@citekey 3] [@citekey, 3] [@citekey p. 3] [@citekey, p. 3] [@citekey page 3] [@citekey, page 3] [@citekey p 3] [@citekey, p 3]`
Probably some other stuff too but since it's written in Haskell and half of the variable names are a single letter, it's a bit hard to tell!

 Nonetheless, *if* we accept that markdown is supposed to be as close as possible to what we might expect to read normally in its plaintext form, it should probably be:
[prefix @citekey, p. 1 suffix] or [prefix @citekey, pp. 1-5 suffix] rather than [@citekey, page 1] or [@citekey, page 1-5]

I don't really know if it's worth bothering with though!

**Comment by retorquere** [comment]
That's easy enough; we could just reuse the mapping table from
scannable-cite.

**Comment by davepwsmith** [comment]
So would the proposed change make it work as such? Sorry if this is just spamming up your repo then close it, I only even heard of coffeescript about three days ago!

**Comment by retorquere** [comment]
Don't worry about spamming, it's a conversation, conversation is good. The
proposed change is A-OK, functionally, it just can't change the label for
all formatters at once.

File: chrome/content/zotero-better-bibtex/cayw.coffee [hunk 3]
```diff
@@ -182,11 +182,12 @@ Zotero.BetterBibTeX.CAYW.Formatter = {
   pandoc: (citations) ->
     formatted = []
     for citation in citations
+      locator = if citation.locator then "#{Zotero.BetterBibTeX.CAYW.shortLocator[citation.label]} #{citation.locator}" else ''
       cite = ''
       cite += "#{citation.prefix} " if citation.prefix
       cite += '-' if citation['suppress-author']
       cite += "@#{citation.citekey}"
-      cite += ", #{citation.label} #{citation.locator}" if citation.locator
+      cite += ", #{locator}"
```
Comment (retorquere) [inline chrome/content/zotero-better-bibtex/cayw.coffee]: This always inserts the comma, even if the locator isn't set. Is this desired? I don't know pandoc well enough to say.

**Comment by retorquere** [comment]
I'm sorry, I've pushed some changes before I say this that conflict with this -- do you want to make the changes to make it mergeable, or do you want me to do it? Attribution is lost when I do it -- it'll be like I made the enhancement, not you.

**Comment by davepwsmith** [comment]
That makes no difference to me really, it's just a learning process for me! There should only be a comma *if* there is a locator. 

On 4 September 2015 00:32:43 BST, retorquere <notifications@github.com> wrote:
>I'm sorry, I've pushed some changes before I say this that conflict
>with this -- do you want to make the changes to make it mergeable, or
>do you want me to do it? Attribution is lost when I do it -- it'll be
>like I made the enhancement, not you.
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/ZotPlus/zotero-better-bibtex/pull/339#issuecomment-137601651

--
https://keybase.io/davidsmith
--
Sent from my phone. Please excuse my brevity.

**Comment by retorquere** [comment]
I don't like to take credit for other peoples' work, but I've integrated the change. A release build for 1.4.0 is running.

**Pull request closed without merging**